features/step_definitions/openid_steps.rb

```diff
@@ -0,0 +1,64 @@
+When /^I register a new client$/ do
+  client_registration_url = "/openid_connect/clients"
+  post client_registration_url,
+       {
+         redirect_uris: ["http://localhost:3000"] # Not actually used
+       }
+end
+
+Given /^I send a post request from that client to the token endpoint using "([^\"]*)"'s credentials$/ do |username|
+  client_json = JSON.parse(last_response.body)
+  user = User.find_by(username: username)
+  token_endpoint_url = "/openid_connect/access_tokens"
+  post token_endpoint_url,
+    {
+      grant_type: "password",
+      username: user.username,
+      password: "password", # Password has been hard coded as all test accounts seem to have a password of "password"
+      client_id: client_json["o_auth_application"]["client_id"],
+      client_secret: client_json["o_auth_application"]["client_secret"]
+    }
+end
+
+Given /^I send a post request from that client to the token endpoint using invalid credentials$/ do
+  client_json = JSON.parse(last_response.body)
+  token_endpoint_url = "/openid_connect/access_tokens"
+  post token_endpoint_url,
+       {
+         grant_type: "password",
+         username: User.find_by(username: "bob"),
+         password: "wrongpassword",
+         client_id: client_json["o_auth_application"]["client_id"],
+         client_secret: client_json["o_auth_application"]["client_secret"]
+       }
+end
+
+When /^I use received valid bearer tokens to access user info$/ do
+  access_token_json = JSON.parse(last_response.body)
+  user_info_endpoint_url = "/api/v0/user/"
+  get user_info_endpoint_url,
+    {
```

Comment:
Align the parameters of a method call if they span more than one line.<br>Redundant curly braces around a hash parameter.